### PR TITLE
Dh roi ij

### PR DESCRIPTION
**Pull request by DavidHerreros**
`protocol_roiIJ` added into `scipion-em-xmipp`. This protocol implements the possibility of drawing Region of Interests in a given Tomogram, so they can be handled in Scipion.

This protocol required the changes from [https://github.com/scipion-em/scipion-em-tomo/pull/27](https://github.com/scipion-em/scipion-em-tomo/pull/27#issue-349306399) to run properly.

### CODE REVIEW

File: xmipp3/protocols/__init__.py [hunk 1]
```diff
@@ -133,3 +133,8 @@
         'To use a Tomography protocol scipion-em-tomo plugin is required.'
         ' See https://github.com/scipion-em/scipion-em-tomo for further details')
 
+from .protocol_roiIJ import XmippProtRoiIJ
+import pyworkflow
+
+emprotocol = pyworkflow.em.protocol.EMProtocol
+setattr(emprotocol, "_createSetOfMeshes", XmippProtRoiIJ._createSetOfMeshes.__func__)
```
Comment (dmaluenda) [inline xmipp3/protocols/__init__.py]: Why is this needed here instead of inside the protocol who needs?

File: xmipp3/protocols/protocol_roiIJ.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+# *  e-mail address 'scipion@cnb.csic.es'
+# *
+# **************************************************************************
+import os
+
+from pyworkflow.em import *
+from pyworkflow.em.viewers.showj import *
+
+from tomo.objects import Mesh, SetOfMeshes
+from tomo.viewers.views_tkinter_tree import TomogramsTreeProvider, TomogramsDialog
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_roiIJ.py]: Please, to import from another plugin use the following
```
from pyworkflow.utils import pluginNotFound
try:
    from tomo.objects import Mesh, SetOfMeshes
    from tomo.viewers.views_tkinter_tree import TomogramsTreeProvider, TomogramsDialog
except ImportError:
    pluginNotFound("tomo")
```
in order to avoid exceptions when plugin not found.

When a single (or a few) imports are needed, use this
```
from pyworkflow.utils import importFromPlugin
myFunction = importFromPlugin("tomo.whatever", "myFunction")
```

File: xmipp3/protocols/__init__.py (hunk 1 above)
Comment (DavidHerreros) [reply xmipp3/protocols/__init__.py]: The `setattr` is included in the `__init__.py` so the method `_createSetOfMeshes` can be used by the create subset and user subset programs.
